src/main/java/org/team199/robot2022/subsystems/Climber.java

```diff
@@ -18,41 +21,63 @@ public class Climber extends SubsystemBase {
-    private static final double retractPositionRight = -1.172;
     private static final double gearing = 9;
     private static final double kInPerSec = ((kNEOFreeSpeedRPM / gearing) * Math.PI * kDiameterIn / 60);
-    private static double kRetractSpeed = -((kDesiredRetractSpeedInps / kInPerSec)
-            + (kVoltsToCounterTorque / 12)); // ~ -0.06151
-    private static double kExtendSpeed = (kDesiredExtendSpeedInps / kInPerSec); // ~0.30261
-
     private static final double kSlowDesiredRetractSpeedInps = 1;
     private static final double kSlowDesiredExtendSpeedInps = 1;
 
     // Torque is 2 * 9 * 0.5 = 9
     // Torque on the motor is Torque / ( gearing = 9 ) = 1
-
+    private static final double kVoltsToCounterTorque = 10.5;
     private static final double kSlowVoltsToCounterTorque = (1.1D / 32) * 12;
-    private static final double kSlowRetractSpeed = -((kSlowDesiredRetractSpeedInps / kInPerSec)
-            + (kSlowVoltsToCounterTorque / 12)); // ~ -0.06151
-    private static final double kSlowExtendSpeed = (kSlowDesiredExtendSpeedInps / kInPerSec); // ~0.30261
+
+    public static enum MotorSpeed{retract,extend,slowRetract,slowExtend;};
+    private static Dictionary dMotorSpeed = new Hashtable();//given values in constructor
+
+
+    public static enum EncoderPos{extendLeft,extendRight,retractLeft,retractRight,zero;};
+    private static Dictionary dEncoderPos = new Hashtable();//given values in constructor
+
+
+    public static final int leftMotor = -1;//because someone requested this and there's no point slapping them in an enum
+    public static final int bothMotors = 0;
+    public static final int rightMotor = 1;
+
 
     private final CANSparkMax left = MotorControllerFactory.createSparkMax(Constants.DrivePorts.kClimberLeft);
     private final CANSparkMax right = MotorControllerFactory.createSparkMax(Constants.DrivePorts.kClimberRight);
     private final RelativeEncoder leftEncoder = left.getEncoder();
     private final RelativeEncoder rightEncoder = right.getEncoder();
 
+
+    private final Consumer[] setEncoder = new Consumer[]{ // faster to array[](inp) than a bunch of if's in a func
+      (pos) -> leftEncoder.setPosition((double) pos),
+      (pos) -> {
+        leftEncoder.setPosition((double) pos);
+        rightEncoder.setPosition((double) pos);
+      },
+      (pos) -> rightEncoder.setPosition((double) pos),
+    };
+
+    private final Consumer[] setMotor = new Consumer[]{
+      (speed) -> {
+        left.set((double) speed);
+        SmartDashboard.putString("Left Climber State", "Moving");
+      },
+      (speed) -> {
+        left.set((double) speed);
+        right.set((double) speed);
+        SmartDashboard.putString("Left Climber State", "Moving");
+        SmartDashboard.putString("Right Climber State", "Moving");
+      },
+      (speed) -> {
+        right.set((double) speed);
+        SmartDashboard.putString("Right Climber State", "Moving");
+      },
+    };
+
```

Comment:
Please avoid raw types. Use [`DoubleConsumer`](https://docs.oracle.com/javase/8/docs/api/java/util/function/DoubleConsumer.html) instead. This should remove the need for a cast to `double`.

Reply:
Oh, neat.
Didn't know that was a class.